crates/matrix-sdk-crypto/src/machine.rs

```diff
@@ -563,19 +563,37 @@ impl OlmMachine {
             );
         };
 
-        match &event.content {
-            RoomKeyContent::MegolmV1AesSha2(content) => {
-                let session = InboundGroupSession::new(
-                    sender_key,
-                    event.keys.ed25519,
-                    &content.room_id,
-                    &content.session_key,
-                    event.content.algorithm(),
-                    None,
+        let handle_key = |content: &MegolmV1AesSha2Content| {
+            let session = InboundGroupSession::new(
+                sender_key,
+                event.keys.ed25519,
+                &content.room_id,
+                &content.session_key,
+                event.content.algorithm(),
+                None,
+            );
+
+            if let Ok(session) = session {
+                info!(
+                    sender = %event.sender,
+                    sender_key = sender_key.to_base64(),
+                    room_id = %content.room_id,
+                    session_id = session.session_id(),
+                    algorithm = %event.content.algorithm(),
+                    "Received a new megolm room key",
                 );
 
                 Ok(Some(session))
+            } else {
+                unsupported_warning();
+                Ok(None)
             }
+        };
+
+        match &event.content {
+            RoomKeyContent::MegolmV1AesSha2(content) => handle_key(content),
+            #[cfg(feature = "experimental-algorithms")]
+            RoomKeyContent::MegolmV2AesSha2(content) => handle_key(content),
```

Comment:
I'm surprised `content` is of type `MegolmV1AesSha2Content` here (as `handle_key` will expect), must check.

Reply:
Checked: `MegolmV2AesSha2Content` is a type alias to `MegolmV1AesSha2Content` as defined in https://github.com/matrix-org/matrix-rust-sdk/pull/951/files#diff-e746e9dee91391967b6d55fca252ef46ddbd25f4b0d76e95ddf783094b7779deR138, so it all works :-).